### ISSUE TITLE
Use refresh_token to refresh access_token

### ISSUE THREAD

**Issue by mnmt** [mnmt/Dropbox_Download_Convert_Images#5]
Can make use of `oauth_result.expires_at` (a `datetime.datetime()` instance) to recognise when refresh is needed. 

Examples seen at
https://github.com/dropbox/dropbox-sdk-python/blob/main/example/oauth/commandline-oauth-pkce.py
https://github.com/dropbox/dropbox-sdk-python/blob/main/example/oauth/commandline-oauth-scopes.py
though not so clear on how the refresh token is used, and whether the access token is then meant to be extracted and cached until it too expires.

Some useful info on:
https://www.dropboxforum.com/t5/Dropbox-API-Support-Feedback/Oauth2-refresh-token-question-what-happens-when-the-refresh/m-p/499772/highlight/true#M24808

Also see documentation:  
- https://dropbox-sdk-python.readthedocs.io/en/latest/api/dropbox.html#dropbox.dropbox_client.Dropbox.__init__
- https://dropbox-sdk-python.readthedocs.io/en/latest/api/dropbox.html#dropbox.dropbox_client.DropboxTeam.check_and_refresh_access_token 
- https://dropbox-sdk-python.readthedocs.io/en/latest/api/dropbox.html#dropbox.dropbox_client.DropboxTeam.refresh_access_token

Source code:  
- check_and_refresh_access_token source code: https://github.com/dropbox/dropbox-sdk-python/blob/f3f8729ebd1ad7824509890e37402af91b2b6255/dropbox/dropbox_client.py#L360

**Comment by mnmt** [comment]
Dropbox's SDK for its API seems to encourage only using the `refresh_token` for authenticating, beyond the very first auth where neither the `access_token` nor the `refresh_token` exists yet. 

The first auth provides an `access_token` and _the_ `refresh token`, where the `access_token` eventually expires while the `refresh token` never does. 
The access_token alone can be used to authenticate a `dropbox.Dropbox(oauth2_access_token=access_token)` instance while the `access_token` remains valid. However, any `dropbox.Dropbox()` instance including the first one ever generated can be generated by instead calling `dropbox.Dropbox(oauth2_refresh_token=refresh_token, app_key = app_key)`. This is shown in the official example [commandline-oauth-pkce.py](https://github.com/dropbox/dropbox-sdk-python/blob/main/example/oauth/commandline-oauth-pkce.py), where the initial access_token obtained in `oauth_result` is never used.
 
When any method accessing the API is called, the `self.request()` method is called, which in turn calls the `self.check_and_refresh_access_token()` method. This method produces a new `access_token` if `(needs_refresh or needs_token) and can_refresh` is True ([source code](https://github.com/dropbox/dropbox-sdk-python/blob/f3f8729ebd1ad7824509890e37402af91b2b6255/dropbox/dropbox_client.py#L360)). Naturally, a new `access_token` is always generated if no `access_token` was originally provided. 

The class produces a new access_token using `self.refresh_access_token()` ([source code](https://github.com/dropbox/dropbox-sdk-python/blob/f3f8729ebd1ad7824509890e37402af91b2b6255/dropbox/dropbox_client.py#L374)), which does not return the `access_token` directly and instead modifies the `self._oauth2_access_token` attribute to add or update it. By convention, this `access_token` is not meant to be extracted directly ([PEP-8 on internal interfaces](https://peps.python.org/pep-0008/#public-and-internal-interfaces)), and there are no public interfaces for obtaining this information given. 

In conclusion, the code here should be changed to only use the `refresh_token` and `app_key` when calling `dropbox.Dropbox()` as in [commandline-oauth-pkce.py](https://github.com/dropbox/dropbox-sdk-python/blob/main/example/oauth/commandline-oauth-pkce.py). This will mean a new `access_token` is generated every time a new instance is generated, so may be sensible to try to re-use any instance once created, within a single session.
